pkg/hash_cache.go

```diff
@@ -502,6 +487,29 @@ func hashRule(thc *TargetHashCache, rule *build.Rule, configuration *analysis.Co
 	ownConfiguration := NormalizeConfiguration(configuration.GetChecksum())
 
 	// Hash rule inputs
+	labelsAndConfigurations, err := getConfiguredRuleInputs(thc, rule, ownConfiguration)
+	if err != nil {
+		return nil, err
+	}
+	for _, ruleInputLabelAndConfigurations := range labelsAndConfigurations {
+		for _, ruleInputConfiguration := range ruleInputLabelAndConfigurations.Configurations {
+			ruleInputLabel := ruleInputLabelAndConfigurations.Label
+			ruleInputHash, err := thc.Hash(LabelAndConfiguration{Label: ruleInputLabel, Configuration: ruleInputConfiguration})
+			if err != nil {
+				return nil, fmt.Errorf("failed to hash configuredRuleInput %s %s which is a dependency of %s %s: %w", ruleInputLabel, ruleInputConfiguration, rule.GetName(), configuration.GetChecksum(), err)
+			}
+
+			writeLabel(hasher, ruleInputLabel)
+			hasher.Write(ruleInputConfiguration.ForHashing())
+			hasher.Write(ruleInputHash)
```

Comment:
Looks like we're writing the label twice here. Maybe we can dedupe?

Reply:
Where's the other write? I see one on line 502, but `ruleInputHash` doesn't itself hash `ruleInputLabel`, it just uses the label as a cache lookup key?

Reply:
My bad, I misread `ruleInputHash` for `ruleInputLabel`... 👍